RUFAS/routines/feed_storage/storage.py

```diff
@@ -35,24 +42,58 @@ class Storage:
         Recalculates the relative nutrient concentrations after dry matter loss.
     """
 
-    def __init__(self):
+    def __init__(self, capacity: float = float("inf")):
+        self.acceptable_crops: List[CropCategory] = []
+        self.capacity = capacity
         self.stored: List[HarvestedCrop] = []
-        self.capacity = float("inf")
 
-    def receive_crop(self, crop: HarvestedCrop) -> None:
+    @property
+    def stored_mass(self) -> float:
+        """The total mass (kg) of currently stored crops"""
+        return sum(crop.fresh_mass for crop in self.stored)
+
+    def receive_crop(self, crop: HarvestedCrop, time: Time) -> None:
         """
         Receives a harvested crop and adds it to the storage.
 
         Parameters
         ----------
         crop : HarvestedCrop
             The harvested crop to be added to the storage.
+        time : Time
+            An instance of the Time class, needed to set storage time.
 
         Returns
         -------
         None
-        """
-        pass
+
+        Raises
+        ------
+        NotImplementedError
+            If the storage's acceptable crops is not populated.
+        ValueError
+            If the crop's category is not compatible with the storage.
+        Exception
+            If adding the crop exceeds the storage's capacity.
+
+        """
+        if not self.acceptable_crops:
+            raise NotImplementedError(
+                "Storage.acceptable_crops is not populated, consider populating it in the child class."
+            )
+        if crop.category not in self.acceptable_crops:
+            raise ValueError(
+                f"Can't recieve the crop, the compatible crop categories are {self.acceptable_crops=},\
+                    {crop.category} is not one of them."
+            )
+        if self.stored_mass + crop.fresh_mass > self.capacity:
```

Comment:
```suggestion
        unacceptable_crop = crop.category not in self.acceptable_crops
        if unacceptable_crop:
            raise ValueError(
                f"Can't recieve the crop, the compatible crop categories are {self.acceptable_crops=},\
                    {crop.category} is not one of them."
            )
        capacity_exceeded = self.stored_mass + crop.fresh_mass > self.capacity
        if capacity_exceeded:
```
Missed these the first time through. Up to you if they are strictly necessary though.

Reply:
I have mentioned this in some other PRs before: we use documentary variables when understanding the statement is difficult. Here, the next line is explaining what is happening. Putting every single statement in variables is not a good practice.